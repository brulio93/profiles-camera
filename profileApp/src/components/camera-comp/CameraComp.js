import React, { Component } from 'react';
import { Dimensions, StyleSheet, Text, View, TouchableHighlight } from 'react-native';
import Camera from 'react-native-camera';
import styles from '../../styles/Styles';
import {inject, observer} from 'mobx-react';

@inject(['ProfileStore'])
@observer
export default class CameraComp extends React.Component{
    static navigationOptions = ({navigation}) => ({
        title: 'Camera',
        headerBackTitle: null
    })

    takePicture(){
        const options = {};
        this.camera.capture({metadata: options})
          .then((data) => {
                this.props.ProfileStore.photo_uri = data.path;
                this.props.toggleFunction();
                return console.log(data)
            })
          .catch(err => console.error(err));
    }

    render(){
        return(
            <View style={{flex: 1}}>
                <View style={styles.cameraContainer}>
                    <Camera ref={(cam) => {this.camera = cam}} style={styles.preview} aspect={Camera.constants.Aspect.fill}>                    
                    </Camera>
                </View>              
                
                <View>
                    <TouchableHighlight style={styles.capturePic} onPress = {this.takePicture.bind(this)}>
                        <Text style={styles.captureText}>Take a Photo</Text>                                             
                    </TouchableHighlight>      
                </View>
            </View>
        );
    }
}