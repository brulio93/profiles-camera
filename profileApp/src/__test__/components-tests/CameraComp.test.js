import 'react-native';
import React from 'react';
import CameraComp from '../../components/camera-comp/CameraComp';
import renderer from 'react-test-renderer';

const ProfileStore = require('../../stores/ProfileStore').default;

jest.mock('react-native-camera', () => require.requireActual('../../__mocks__/Camera').default);
describe('Camera component', () => {
    test('create camera correctly', () => {
        const render = renderer.create(
            <CameraComp ProfileStore={ProfileStore.ProfileStore}/>
        ).toJSON();
        expect(render).toMatchSnapshot();
    });
});
