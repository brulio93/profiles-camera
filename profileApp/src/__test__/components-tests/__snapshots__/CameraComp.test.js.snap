// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera component create camera correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "height": 5,
      }
    }
  />
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "backgroundColor": "#517fa4",
            "borderColor": "#fff",
            "borderWidth": 1,
            "marginBottom": 40,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 10,
            "padding": 10,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "textAlign": "center",
          }
        }
      >
        Take a Photo
      </Text>
    </View>
  </View>
</View>
`;
